pkg/testing/fetcher_artifact.go

```diff
@@ -68,6 +68,7 @@ func (f *artifactFetcher) Fetch(ctx context.Context, operatingSystem string, arc
 		uri, prevErr = findURI(ctx, f.doer, version)
 	}
 	preVersion := version
+	version, _ = splitBuildID(version)
 	if uri == "" {
 		if !strings.HasSuffix(version, "-SNAPSHOT") {
```

Comment:
If we have a snapshot with a buildId we need to prepare the uri in the format of x.y.z-<buildid>, for example
`https://snapshots.elastic.co/8.11.0-2574a657/downloads/beats/elastic-agent/elastic-agent-8.11.0-SNAPSHOT-linux-x86_64.tar.gz` (we can see the build details from https://artifacts-api.elastic.co/v1/versions/8.11.0-SNAPSHOT/builds/8.11.0-2574a657 )